apollo-runtime/src/main/java/com/apollographql/apollo/internal/InternalUtils.java

```diff
@@ -0,0 +1,33 @@
+package com.apollographql.apollo.internal;
+
+import java.util.Set;
+
+class InternalUtils {
+
+    /**
+     * Checks if two {@link Set} are disjoint. Returns true if the sets don't have a single common element. Also returns
+     * true if either of the sets is null.
+     *
+     * @param setOne the first set
+     * @param setTwo the second set
+     * @param <E>    the value type contained within the sets
+     * @return True if the sets don't have a single common element or if either of the sets is null.
+     */
+    static <E> boolean areDisjoint(Set<E> setOne, Set<E> setTwo) {
```

Comment:
Moved this here instead of converting to Kotlin since it was only used in `runtime` module.